mock/src/main/java/com/tngtech/keycloakmock/impl/helper/RedirectHelper.java

```diff
@@ -42,9 +44,16 @@ public String getRedirectLocation(
     }
 
     ResponseMode responseMode = responseType.getValidResponseMode(session.getResponseMode());
-    StringBuilder redirectUri = new StringBuilder(session.getRedirectUri());
-    redirectUri.append(getResponseParameter(responseMode, STATE, session.getState()));
-    redirectUri.append(getResponseParameter(null, SESSION_STATE, session.getSessionId()));
+    String originalRedirectUri = session.getRedirectUri();
+    StringBuilder redirectUri;
+    if (OOB_REDIRECT.equals(originalRedirectUri)) {
+      redirectUri =
+          new StringBuilder(requestConfiguration.getOutOfBandLoginLoginEndpoint().toASCIIString());
+    } else {
+      redirectUri = new StringBuilder(originalRedirectUri);
+    }
+    redirectUri.append(getResponseParameter(responseMode, SESSION_STATE, session.getSessionId()));
+    redirectUri.append(getResponseParameter(null, STATE, session.getState()));
```

Comment:
Are the parameters here changed by accident?

Reply:
No, I switched them because the state is now optional, but the first parameter generates the ? or #. This would be skipped if the parameter was null, so I used a mandatory parameter as the first parameter.